src/wp-includes/block-template-utils.php

```diff
@@ -724,6 +724,74 @@ function _wp_build_title_and_description_for_taxonomy_block_template( $taxonomy,
 	return true;
 }
 
+/**
+ * Build a block template object for based on a WP_Post object.
+ *
+ * @since 6.5.0
+ * @access private
+ *
+ * @param WP_Post $post Template post.
+ * @param array $additional_fields Additional fields to add to the template object.
+ * @param bool $hook_blocks Whether to insert hooked blocks into the content or not.
+ * @return WP_Block_Template|WP_Error Template or error object.
+ */
+function _build_block_template_object_from_wp_post_object( $post, $additional_fields = array() ) {
+	$default_template_types = get_default_block_template_types();
+
+	if ( empty( $additional_fields['theme'] ) ) {
+		return new WP_Error( 'template_missing_theme', __( 'No theme is defined for this template.' ) );
+	}
+	$theme = $additional_fields['theme'];
+
+	$template_file  = _get_block_template_file( $post->post_type, $post->post_name );
+	$has_theme_file = get_stylesheet() === $theme && null !== $template_file;
+
+	/*
+	 * These are fields that are directly mapped from the post object to the template object.
+	 */
+	$post_to_template_mapping = array(
+		'ID'            => 'wp_id',
+		'post_content'  => 'content',
+		'post_name'     => 'slug',
+		'post_type'     => 'type',
+		'post_excerpt'  => 'description',
+		'post_title'    => 'title',
+		'post_status'   => 'status',
+		'post_author'   => 'author',
+		'post_modified' => 'modified',
+	);
+
+	$template = new WP_Block_Template();
+
+	foreach ( $post_to_template_mapping as $post_key => $template_key ) {
+		if ( isset( $post->{$post_key} ) && property_exists( $template, $template_key ) ) {
+			$template->{$template_key} = $post->{$post_key};
+		}
+	}
+
+	if ( is_array( $additional_fields ) && ! empty( $additional_fields ) ) {
+		foreach ( $additional_fields as $key => $value ) {
+			if ( property_exists( $template, $key ) ) {
+				$template->{$key} = $value;	
+			}
+		}
+	}
+
+	$template->id = ! empty( $template->theme ) && ! empty( $template->slug ) ? $template->theme . '//' . $template->slug : null;
```

Comment:
AFAICS, we haven't set `$template->theme` at this point yet (only `$theme`) 🙈 

I _might_ make some of the mappings a bit more verbose again; maybe it'll make them a bit easier to track.

Reply:
We should have since the [`$additional_fields` mapping is run prior to it](https://github.com/WordPress/wordpress-develop/pull/6278/files#diff-1d8e05d3ea69557ca8742330e62c1893a72d4ca4ff2ed4ccc5b94bcf5d20fed7R772-R778) which at this point should have it present because it would have otherwise[ returned an error.](https://github.com/WordPress/wordpress-develop/pull/6278/files#diff-1d8e05d3ea69557ca8742330e62c1893a72d4ca4ff2ed4ccc5b94bcf5d20fed7R741-R743)

Reply:
`homer-disappear.gif`

Reply:
Haha! I'm pushing an update to make this more clear!

Reply:
https://github.com/WordPress/wordpress-develop/pull/6278/commits/adcec7a335012d060126d0fca6527943cfe6d833